graphql/server.go

```diff
@@ -344,8 +387,17 @@ func Handler(schema *Schema) http.Handler {
 	})
 }
 
+func (c *conn) Use(fn MiddlewareFunc) {
+	c.middlewares = append(c.middlewares, fn)
+}
+
 func ServeJSONSocket(ctx context.Context, socket JSONSocket, schema *Schema, makeCtx MakeCtxFunc, logger GraphqlLogger) {
```

Comment:
I find the "Use" function name to be to generic to be helpful. I think breaking ServeJSONSocket is fine. We'll break it again when we drop the makeCtx and logger endpoints (which I think is a great idea.)